src/block/Sugarcane.php

```diff
@@ -98,6 +98,12 @@ public function ticksRandomly() : bool{
 
 	public function onRandomTick() : void{
 		if(!$this->getSide(Facing::DOWN)->hasSameTypeId($this)){
+			if(!$this->isSupportedByWater()) {
```

Comment:
This should happen not only during onRandomTick but also if something nearby is updated, for example, when a block is placed nearby.

https://github.com/user-attachments/assets/d6becf94-1c18-46d3-aae2-ecfc2c13fd4a

Reply:
I added onNearbyBlockChange for this thanks 👍